### DISCUSSION TITLE
Wallet V2

### DISCUSSION THREAD

**Discussion by bradleystachurski** [fedimint/fedimint#5893] [Fedimint Modules]
Capturing an out-of-band conversation with @joschisan: let's use this thread to iterate on designs for wallet v2. For previous designs, please refer to https://github.com/fedimint/fedimint/blob/master/docs/wallet_module_v2.md

**Comment by dpc** [comment]
> This document outlines the design for wallet v2, a ground up redesign similar in nature to lnv2.

![image](https://github.com/user-attachments/assets/6868f2c2-9965-4fdc-9e41-d9b701cd53eb)

I don't see any document.

**Reply by bradleystachurski** [reply]
😄 That's funny, thanks for the ping. Typing while talking is dangerous. Hopefully the updated comment is less confusing.

**Comment by joschisan** [comment]
Currently we have the following improvements in mind:

The current wallet design sets a very high fee rate in order to assume that every transaction by the federation is confirmed. Rbf is already deactivated and would be removed for a v2. Hence we can immediately use change outputs of transactions and do not need to wait for them to be confirmed. Therefore all assets of the federation are always available for pegouts.

Assuming that every transaction is confirmed we can also create a consolidation transaction on every pegin. This would lead to the federation holding only a single utxo. The fees for the consolidation transaction will be subtracted from the pegin utxo.

Finally we can also switch to naive taproot threshold signatures.

**Comment by bradleystachurski** [comment]
Call: in wallet v2, we should use block height instead of block count for consistency
